### ISSUE TITLE
Question: Cordova hybrid app - is it possible?

### ISSUE THREAD

**Issue by theDude30** [twilio/twilio-video.js#85]
Hi,
I'm trying to create hybrid application that should load the twilio Javascript and enable audio/video.
Is it possible? did anyone had any experience with that?

**Comment by markandrus** [comment]
Hi @theDude30,

We don't support this directly today, although I know others have been experimenting with it. We're currently focusing on 1.0.0, but we should have future opportunities to explore these frameworks.

Best,
Mark

**Comment by drbeermann** [comment]
We have an Cordova plugin for iOS working and are in the process of testing Android. We had the javascript SDK working in one case but are not sure that it is the most reliable. Right now the audio tracks are not getting created, but I'm unsure why. I know a few others have created an iOS plugin as well. We could potentially open source ours but it isn't the most full-featured right now.

**Comment by theDude30** [comment]
Thanks,
After some struggling i ended using the twilio iOS SDK + web application(webkit).
as for android I'm using chrome webview(WebChromeClient) and the video works great without any code changes.

**Comment by ErikBleifield** [comment]
A question for the TheDude30 - I am interested in a Cordova interfaces as well, am I understanding this correctly: you could not get Cordova to work, so you had to move to native iOS SDK??

**Comment by PriceFallin** [comment]
My team made a plugin for the latest Twilio Video SDK that acts as a Cordova plugin for our Ionic App. We have it working great on iOS and Android. A while back we tried getting the JS Twilio SDK to work in our hybrid app but didn't have much luck, even with the crosswalk webview.

**Comment by markandrus** [comment]
@PriceFallin our team would be interested to learn about your Cordova plugin development and the issues you hit with the JavaScript SDK. Would you be willing to chat? If so, can you email me at mroberts@twilio.com?

**Comment by ErikBleifield** [comment]
@PriceFallin - asking the obvious follow up, is the plugin anything that can be made public or shareable?

**Comment by PriceFallin** [comment]
Hey @ErikBleifield, talking to the Twilio team about it this week. Stay tuned. I'll post updates.

**Comment by vinlim** [comment]
Hi, following the matter, is there any usable plugin for twilio video now?

**Comment by PriceFallin** [comment]
@strangecyan 
I reached out to the Twilio team to follow up. If I don't hear back soon I'll try to create a github repo for it. There are some custom status notifications relevant to my project which I will need to remove first. Please don't count on any support for it and documentation will also be limited. If I update the code I'll push a new build but don't have time to manage a public repo (which is why I reached out to Twilio to see if they would).

**Comment by strangecyan** [comment]
@PriceFallin Thank you! I’ll be anxiously waiting, I’ve got to start on something tomorrow using Twilio video and I’m unsure whether to redo all the work that you’ve done or wait until you get the chance to strip out the sensitive bits. Would you feel more comfortable just sharing privately?

**Comment by jcuervo** [comment]
Thanks @PriceFallin! pls update this thread. I know a lot have been waiting patiently.

**Comment by Intraview-in** [comment]
Hi, Would be great if someone shares the input here. We are also anxiously waiting for Cordova support in Twilio Video API's.

**Comment by PriceFallin** [comment]
Thank you all for being patient. I had a two week sprint to finish up before a production release. I've been working hard on getting this plugin all set up in a github public repo. I did get a hold of twilio but it doesn't sound like supporting a cordova plugin for their video chat system is high on their priority list.

I hope you all are able to get this plugin working for your needs/interest. I was not able to test it with the changes I made to the alerts or renaming of files. Feel free to create 'issues' if you have them and I'll try to respond when I get time.

Here's the repo: [https://github.com/PriceFallin/cordova-plugin-twilio-video](url)
